### ISSUE TITLE
[Bug]: The `ApiException` exception thrown by `EnsureSuccessStatusCodeAsync()` is always missing a value for the `Content` property

### ISSUE THREAD

**Issue by nathanpovo** [reactiveui/refit#1465]
### Describe the bug 🐞

When calling the `EnsureSuccessStatusCodeAsync()` method on an unsuccessful request, the `Content` property of the `ApiException` that is thrown is always `null` **even if the `Content` property of the `Error` property of the response has a value** (shouldn't both errors have the same value?).

### Step to reproduce

Give the following client

```csharp
public interface IApi {
    [Get("/500")]
    Task<ApiResponse<string>> Get500Error();
}
```

Call it using:

```csharp
var client = RestService.For<IApi>("https://httpstat.us");

using ApiResponse<string> response = await client.Get500Error();
```

And ensure that the response was successful

```csharp
await response.EnsureSuccessStatusCodeAsync();
```

### Reproduction repository

https://gist.github.com/nathanpovo/082104fba9cef5923017b398786a3697

### Expected behavior

The exception that is thrown by `EnsureSuccessStatusCodeAsync()` should have a value for the `Content` property.

### Actual behavior

The `Content` property of the exception that is thrown by `EnsureSuccessStatusCodeAsync()` is always `null`.

### Workaround

Instead of using `EnsureSuccessStatusCodeAsync()`, manually check that the response is unsuccessful and throw the exception yourself:

```diff
-await response.EnsureSuccessStatusCodeAsync();
+if (!response.IsSuccessStatusCode)
+{
+    throw response.Error;
+}
```

### Screenshots 🖼️

Exception with content:

![image](https://user-images.githubusercontent.com/17831160/211149759-e9fdd9a5-2e33-497c-94e3-07a20d333221.png)

Exception with no content:

![image](https://user-images.githubusercontent.com/17831160/211149788-cd86e8eb-007e-42db-964c-1871fed743c2.png)

### IDE

Rider Windows

### Operating system

Windows

### ReactiveUI Version

6.3.2

### Additional information ℹ️

Related to #1221

**Comment by marcominerva** [comment]
I think this is related to: https://github.com/reactiveui/refit/issues/1376. I have already provided a fix, hoping it will be eventually merged.

**Comment by nathanpovo** [comment]
The issue seems to be due to the `Error` property of the response already being populated with the exception before `EnsureSuccessStatusCodeAsync()` is called.

`EnsureSuccessStatusCodeAsync()` creates a new exception when the response is not successful

https://github.com/reactiveui/refit/blob/246ee8d9989c29092fdaba6821d5b3098a5ccf9e/Refit/ApiResponse.cs#L85-L97

The exception is created with the following code

https://github.com/reactiveui/refit/blob/246ee8d9989c29092fdaba6821d5b3098a5ccf9e/Refit/ApiException.cs#L119-L150

The problem is that, since the exception was already created, the `HttpConnectionResponseContent` of the response has already been disposed by the time `EnsureSuccessStatusCodeAsync()` is called

![image](https://user-images.githubusercontent.com/17831160/211150656-d8866ed2-02bc-4275-a81e-fd2526f2ce6a.png)

And thus the `Content` of the response is always `null`.

**Comment by nathanpovo** [comment]
> I think this is related to: #1376. I have already provided a fix, hoping it will be eventually merged.

@marcominerva I did not see your issue while investigating my own problem. You are right, your fix will also solve this bug.

We were experiencing different symptoms of the same root cause.

**Comment by nathanpovo** [comment]
Fixed with #1448

Upgrade to version [6.4.1](https://github.com/reactiveui/refit/releases/tag/6.4.1) to get the fix.

**Comment by ChrisPulman** [comment]
> Fixed with #1448
> 
> Upgrade to version [6.4.1](https://github.com/reactiveui/refit/releases/tag/6.4.1) to get the fix.

Due to an unexpected reason you may need to manually set the version of the package [6.4.1] and [6.5.1] have somehow been unlisted but they do exist.
